server/src-lib/Hasura/Server/Query.hs

```diff
@@ -304,10 +307,7 @@ getQueryAccessMode q = (fromMaybe Q.ReadOnly) <$> getQueryAccessMode' q
             "incompatible access mode requirements in bulk query, " <>
             "expected access mode: " <>
             (T.pack $ maybe "ANY" show expectedMode) <>
-            " but " <>
-            "$.args[" <>
-            (T.pack $ show i) <>
-            "] forces " <>
+            " but " <> "$.args[" <> (T.pack $ show i) <> "] forces " <>
```

Comment:
We don't have to manually construct the path, it'll be handled automatically with `withPathK` and `indexedFoldM` above, just the error message will be sufficient.